sdk/core/System.ClientModel/src/Convenience/ClientResult.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.ClientModel.Primitives;
+
+namespace System.ClientModel;
+
+public abstract class ClientResult
```

Comment:
@KrzysztofCwalina: why is this abstract given no members are abstract?
https://github.com/Azure/azure-sdk-for-net/pull/41009#discussion_r1442253842

Reply:
This follows the pattern in Azure.Core where we have abstract `Response`, `Response<T>` and private implementations internally.  I wanted to be consistent with Azure.Core for all the result types.  If this is a problem, let's discuss the practical implications.

Reply:
Response and Response<T> have abstract members, and so they have to be abstract. This type does not have any abstract members. The negative implications of this current design are: 1) you need the internal subclass which would not be necessary if this base class was not abstract and so non-abstract ClientResult is simpler. 2) marking a class abstract is a message/hint to the user about how the type works. Here this hint is misleading.

Reply:
Thank you for the explanation!  `Response<T>` doesn't have abstract members, but you're right that `NullableResponse<T>` does.  I can change this to be non-abstract.

Reply:
Since we've made the properties on `OptionalClientResult<T>` virtual, given the simplification argument and the UX argument, it will probably be better to make the other ClientResult types non-abstract as well?  I'll change that as part of the ClientResult change, but let me know if there's a reason not to.

Reply:
> Since we've made the properties on `OptionalClientResult<T>` virtual, given the simplification argument and the UX argument, it will probably be better to make the other ClientResult types non-abstract as well? I'll change that as part of the ClientResult change, but let me know if there's a reason not to.

Yeah, I think only the pipeline reques/response need to be abstract.

Reply:
Addressed.